src/main.rs

```diff
@@ -81,19 +111,57 @@ async fn handle_genesis_public_keys(near_config: near_indexer::NearConfig) {
     while let Some((index, _result)) = insert_genesis_keys.next().await {
         info!(
             target: INDEXER_FOR_WALLET,
-            "Genesis public keys adding {}%",
+            "Dump state public access keys adding {}%",
             index * 100 / total_access_key_chunks
         );
     }
 
-    info!(target: INDEXER_FOR_WALLET, "Genesis public keys handled.");
+    info!(target: INDEXER_FOR_WALLET, "Dumped state public access keys in database successfully replaced.");
+}
+
+async fn insert_receipts(
+    height: near_indexer::near_primitives::types::BlockHeight,
+    chunks: &[near_indexer::near_primitives::views::ChunkView],
+    pool: &Pool<ConnectionManager<PgConnection>>,
+) {
+    let access_keys = chunks
+        .iter()
+        .map(|chunk| &chunk.receipts)
+        .flatten()
+        .filter_map(|receipt| match receipt.receipt {
+            near_indexer::near_primitives::views::ReceiptEnumView::Action { .. } => {
+                Some(AccessKey::from_receipt_view(receipt, height))
+            }
+            _ => None,
+        })
+        .flatten()
+        .collect::<Vec<AccessKey>>();
+
+    info!(
+        target: INDEXER_FOR_WALLET,
+        "Handling receipts related to AccessKey, amount {}",
+        access_keys.len()
+    );
+    if !access_keys.is_empty() {
+        diesel::insert_into(schema::access_keys::table)
+            .values(access_keys)
+            .on_conflict_do_nothing()
+            .execute_async(&pool)
+            .await
+            .unwrap();
+    }
 }
 
 async fn update_receipt_status(
     receipt_ids: Vec<String>,
     status: ExecutionStatus,
     pool: &Pool<ConnectionManager<PgConnection>>,
 ) {
+    info!(target: INDEXER_FOR_WALLET, "update_receipt_status called");
```

Comment:
I think we should promote these (this and the one below) to `debug` or remove completely.

```suggestion
    debug!(target: INDEXER_FOR_WALLET, "update_receipt_status called");
```